pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts

```diff
@@ -0,0 +1,802 @@
+import { range } from 'lodash';
+
+describe('TimelockAuthorizer permissions', () => {
+  let authorizer: TimelockAuthorizer, vault: Contract;
+  let root: SignerWithAddress,
+    nextRoot: SignerWithAddress,
+    revoker: SignerWithAddress,
+    granter: SignerWithAddress,
+    user: SignerWithAddress,
+    other: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, root, nextRoot, granter, revoker, user, other] = await ethers.getSigners();
+  });
+
+  const ACTION_1 = '0x0000000000000000000000000000000000000000000000000000000000000001';
+  const ACTION_2 = '0x0000000000000000000000000000000000000000000000000000000000000002';
+  const ACTION_3 = '0x0000000000000000000000000000000000000000000000000000000000000003';
+
+  const WHERE_1 = ethers.Wallet.createRandom().address;
+  const WHERE_2 = ethers.Wallet.createRandom().address;
+
+  const EVERYWHERE = TimelockAuthorizer.EVERYWHERE;
+  const NOT_WHERE = ethers.Wallet.createRandom().address;
+
+  sharedBeforeEach('deploy authorizer', async () => {
+    let authorizerContract: Contract;
+
+    ({ instance: vault, authorizer: authorizerContract } = await Vault.create({
+      admin: root,
+      nextAdmin: nextRoot.address,
+    }));
+
+    authorizer = new TimelockAuthorizer(authorizerContract, root);
+  });
+
+  describe('grantPermission', () => {
+    context('when there is a delay set to grant permissions', () => {
+      const delay = DAY;
+
+      sharedBeforeEach('set delay', async () => {
+        const setAuthorizerAction = await actionId(vault, 'setAuthorizer');
+        await authorizer.scheduleAndExecuteDelayChange(setAuthorizerAction, delay * 2, { from: root });
+        await authorizer.scheduleAndExecuteGrantDelayChange(ACTION_1, delay, { from: root });
+      });
+
+      it('reverts if requires a schedule', async () => {
+        await expect(authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: root })).to.be.revertedWith(
+          'GRANT_MUST_BE_SCHEDULED'
+        );
+      });
+
+      it('can schedule a grant permission', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: root });
+
+        // should not be able to execute before delay
+        await expect(authorizer.execute(id, { from: root })).to.be.revertedWith('ACTION_NOT_YET_EXECUTABLE');
+
+        await advanceTime(delay);
+        await authorizer.execute(id, { from: root });
+
+        expect(await authorizer.canPerform(ACTION_1, user, WHERE_1)).to.be.true;
+        expect(await authorizer.canPerform(ACTION_2, user, WHERE_2)).to.be.false;
+      });
+    });
+
+    context('when there is a no delay set to grant permissions', () => {
+      it('reverts if the sender is not the granter', async () => {
+        await expect(authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: granter })).to.be.revertedWith(
+          'SENDER_IS_NOT_GRANTER'
+        );
+      });
```

Comment:
This is a bit odd because you're actually using `granter` here, but it's not yet been made a granter. I'd remove this test and check granter status in other places